drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/Downloader.java

```diff
@@ -39,28 +39,30 @@ public static File download(File targetDir, URL from) {
         File target = new File(targetDir + File.separator + fileName);
         File downloaded = null;
 
-        if (target.exists() && target.isFile()) {
-            downloaded = target;
-        } else if (!targetDir.exists()) {
-            targetDir.mkdirs();
-        }
+        synchronized (log) {
+            if (target.exists() && target.isFile()) {
+                downloaded = target;
+            } else if (!targetDir.exists()) {
+                targetDir.mkdirs();
+            }
 
-        if (downloaded == null) {
+            if (downloaded == null) {
 
-            for (int i = 0; i < 3; i++) {
-                try {
-                    downloaded = runDownloadExecution(from, target.getAbsolutePath(), fileName).await();
-                } catch (ExecutionException ee) {
-                    System.err.print("ERROR: the downloading has failed. ");
-                    if (2 - i > 0) {
-                        System.err.println("Trying again - number of remaining attempts: " + (2 - i));
-                        continue;
-                    } else {
-                        System.err.println("For more information see the stacktrace of an exception");
-                        throw ee;
+                for (int i = 0; i < 3; i++) {
+                    try {
+                        downloaded = runDownloadExecution(from, target.getAbsolutePath(), fileName).await();
+                    } catch (ExecutionException ee) {
+                        System.err.print("ERROR: the downloading has failed. ");
+                        if (2 - i > 0) {
```

Comment:
That's somehow cryptic for my taste ;) Maybe we start with `int i = 3` and improve this `if` too?